src/node_rc.cc

```diff
@@ -0,0 +1,198 @@
+  simdjson::ondemand::document document;
+  if (json_parser.iterate(file_content).get(document)) {
+    FPrintF(stderr, "Can't parse %s\n", config_path.c_str());
+    return ParseResult::InvalidContent;
+  }
+
+  simdjson::ondemand::object main_object;
+  // If document is not an object, throw an error.
+  if (auto root_error = document.get_object().get(main_object)) {
+    if (root_error == simdjson::error_code::INCORRECT_TYPE) {
+      FPrintF(stderr,
+              "Root value unexpected not an object for %s\n\n",
+              config_path.c_str());
+    } else {
+      FPrintF(stderr, "Can't parse %s\n", config_path.c_str());
+    }
+    return ParseResult::InvalidContent;
+  }
+
+  ConfigReader::Config config;
+  simdjson::ondemand::value ondemand_value;
+  simdjson::ondemand::raw_json_string key;
+
+  for (auto field : main_object) {
+    if (field.key().get(key) || field.value().get(ondemand_value)) {
+      return ParseResult::InvalidContent;
+    }
+    if (key == "experimental") {
+      simdjson::ondemand::object experimental_object;
+      if (ondemand_value.get_object().get(experimental_object)) {
+        FPrintF(stderr, "Invalid value for experimental object\n");
+        return ParseResult::InvalidContent;
+      }
+      auto result = ParseExperimentalObject(&experimental_object, &config);
+      if (result != ParseResult::Valid) {
+        return result;
+      }
+    }
+
+    if (key == "test") {
+      simdjson::ondemand::object test_object;
+      if (ondemand_value.get_object().get(test_object)) {
+        FPrintF(stderr, "Invalid value for test object\n");
+        return ParseResult::InvalidContent;
+      }
+      auto result = ParseTestObject(&test_object, &config);
+      if (result != ParseResult::Valid) {
+        return result;
+      }
+    }
+
+    if (key == "import") {
+      simdjson::ondemand::array raw_imports;
+      if (ondemand_value.get_array().get(raw_imports)) {
+        FPrintF(stderr, "Invalid value for import\n");
+        return ParseResult::InvalidContent;
+      }
+      for (auto raw_import : raw_imports) {
+        std::string_view import;
+        if (raw_import.get_string(import)) {
+          FPrintF(stderr, "Invalid value for import\n");
+          return ParseResult::InvalidContent;
+        }
+        config.import.push_back(std::string(import));
+      }
+    }
+  }
+
+  config_ = config;
+  return ParseResult::Valid;
+}
+
+void ConfigReader::AssignNodeOptions(std::string* node_options) {
+  std::vector<std::string> result;
+  if (config_.experimental_transform_types.has_value()) {
+    result.push_back(config_.experimental_transform_types.value()
+                         ? "--experimental-transform-types"
+                         : "--no-experimental-transform-types");
+  }
+
+  if (config_.test_only.has_value()) {
+    result.push_back(config_.test_only.value() ? "--test-only"
+                                               : "--no-test-only");
+  }
```

Comment:
This bit is illustrative of why I think it is important to integrate more directly with node_options. It's going to be difficult to keep things appropriately in sync.